utils/Azure/acrTools.ts

```diff
@@ -72,18 +72,30 @@ export async function getRepositoriesByRegistry(registry: Registry): Promise<Rep
     return allRepos;
 }
 
-/** Sends a custon html request to a registry
+/** List images under a specific digest */
+export async function getImagesByDigest(repo: Repository, digest: string): Promise<AzureImage[]> {
+    let allImages: AzureImage[] = [];
+    let image: AzureImage;
+    const { acrAccessToken } = await acquireACRAccessTokenFromRegistry(repo.registry, 'repository:' + repo.name + ':pull');
+    //const tags: TagInfo[] = await getTagAttributes('https://' + element.registry.loginServer, element.name, { bearer: acrAccessToken });
+    const tags: string[] = await getAssociatedTags('https://' + repo.registry.loginServer, repo.name, digest, { bearer: acrAccessToken });
+    for (let tag of tags) {
+        image = new AzureImage(repo, tag);
+        allImages.push(image);
+    }
+    return allImages;
+}
+
+/** Sends a custom html request to a registry
  * @param http_method : the http method, this function currently only uses delete
  * @param login_server: the login server of the registry
  * @param path : the URL path
- * @param username : registry username, can be in generic form of 0's, used to generate authorization header
- * @param password : registry password, can be in form of accessToken, used to generate authorization header
+ * @param accessToken : Bearer access token.
  */
-export async function sendRequestToRegistry(http_method: string, login_server: string, path: string, bearerAccessToken: string): Promise<any> {
+export async function sendRequestToRegistry(http_method: string, login_server: string, path: string, accessToken: string): Promise<any> {
```

Comment:
Do you need to update `<any>` here?

Reply:
I wasn't the author of this and I'll definitely have to change this in order to merge to upstream.
However it looks like this function was expected to do more that just delete, and the it would have to return different types of objects, that's why the used any.
I forgot to ask about this, but If there is no plan to improve this. We should rename it to sendDeleteRequestToRegistry and remove the unnecessary code, or I could just add the only type is returning right now, and keep building on top of that.
What do you prefer?

Reply:
If it's just used by deletion, let's rename it. Long-term, we should be able to use the auto-generated sdk from registry runtime api.

Reply:
It's used only by delete image and repository.
So I'm renaming it

Reply:
I just realized that the http is required anyway. So I'm keeping the name for now and just changing the promise.